src/hxtsdgen/Generator.hx

```diff
@@ -29,7 +31,10 @@ class Generator {
 
         Context.onGenerate(function(types) {
```

Comment:
Can we make the function that collects types in Generator class an instance method, so that it is possible to tweak the generation process in subclasses? Something like this?

```
class MyCustomGenerator extends hxtsdgen.Generator
{
	static function myCustomUse() {
		if (Context.defined("display") || !Context.defined("js"))
			return;

		Context.onGenerate(new Generator().onContextGenerate);
	}

	var exposed = [];
	var includeHeader = false;
	var outJS: String;
	var outDTS: String;

	function new() {
		super();
	}

	public function onContextGenerate(types: Array<Type>) {
		outJS = Compiler.getOutput();
		outDTS = Path.withoutExtension(outJS) + ".d.ts";

		for (type in types) {
			switch [type, type.follow()] {
				case [TType(_.get() => t, _), TAnonymous(_.get() => anon)]:
					if (t.meta.has(":expose") && !t.meta.has(":noexpose")) {
						exposed.push(ETypedef(t, anon));
					}
				case [_, TInst(_.get() => cl, _)]:
					if (cl.meta.has(':enum')) {
						if (cl.meta.has(":expose") && !cl.meta.has(":noexpose")) {
							exposed.push(EEnum(cl));
						}
					} else {
						if (cl.meta.has(":expose") && !cl.meta.has(":noexpose")) {
							exposed.push(EClass(cl));
						}
						for (f in cl.statics.get()) {
							if (f.meta.has(":expose") && !f.meta.has(":noexpose"))
								exposed.push(EMethod(cl, f));
						}
					}
				default:
			}
		}

		includeHeader = !Context.defined("hxtsdgen-skip-header");

		if (exposed.length == 0) {
			var src = hxtsdgen.Generator.NO_EXPOSE_HINT;
			if (includeHeader) src = HEADER + "\n\n" + src;
			sys.io.File.saveContent(outDTS, src);
		} else {
			Context.onAfterGenerate(onContextAfterGenerate);
		}
	}

	function onContextAfterGenerate() {
		var declarations = generate(exposed);

		if (includeHeader)
			declarations.unshift(HEADER);

		sys.io.File.saveContent(outDTS, declarations.join("\n\n"));
	}
```

Reply:
Makes sense but what do you do with that? Maybe there are other ways to inject custom behaviours.

Reply:
I would like to generate TS files and also customize a bit the output like adding lint rules, imports, etc. There is a lot of great functionality that can be reused even more with such possibility.

Reply:
I see. I pushed a big refactoring which should make that very easy to do.

One thing is that the header is now a public static, so you should be able to just change these values maybe with a macro.